Stripe/STPPaymentContext.m

```diff
@@ -626,39 +632,27 @@ - (void)requestPayment {
+            __block BOOL isCanceledAfterPaymentResultCreated = NO;
             STPApplePaySourceHandlerBlock applePaySourceHandler = ^(id<STPSourceProtocol> source, STPErrorBlock completion) {
-                [self.apiAdapter attachSourceToCustomer:source completion:^(NSError *attachSourceError) {
-                    stpDispatchToMainThreadIfNecessary(^{
-                        if (attachSourceError) {
-                            completion(attachSourceError);
-                        } else {
-                            id<STPSourceProtocol> paymentResultSource = source;
-                            /**
-                             When createCardSources is false, the SDK:
-                             1. Sends the token to customers/[id]/sources. This
-                             adds token.card to the customer's sources list.
-                             Surprisingly, attaching token.card to the customer
-                             will fail.
-                             2. Returns token.card to didCreatePaymentResult,
-                             where the user tells their backend to create a charge.
-                             A charge request with the token ID and customer ID
-                             will fail because the token is not linked to the
-                             customer (the card is).
-                             */
-                            if ([source isKindOfClass:[STPToken class]]) {
-                                paymentResultSource = ((STPToken *)source).card;
-                            }
-                            STPPaymentResult *result = [[STPPaymentResult alloc] initWithSource:paymentResultSource];
-                            [self.delegate paymentContext:self didCreatePaymentResult:result completion:^(NSError * error) {
-                                // for Apple Pay, the didFinishWithStatus callback is fired later when Apple Pay VC finishes
-                                if (error) {
-                                    completion(error);
-                                } else {
-                                    completion(nil);
-                                }
-                            }];
+                [self handleApplePaySource:source
+                                completion:^(NSError * error) {
+                    // for Apple Pay, the didFinishWithStatus callback is fired later when Apple Pay VC finishes
+                    // The apple pay vc finishes when the user cancels and it finishes only once.
+                    // So if the user has already canceled, calling completion will not cause the apple pay vc to finish
+                    // Therefore, we must directly call our did finish with status method
+                    if (error) {
+                        completion(error);
+                        if (isCanceledAfterPaymentResultCreated) {
+                            [self didFinishWithStatus:STPPaymentStatusError error:error];
```

Comment:
Your documentation on the delegate said the delegate method `paymentContext:didFinishWithStatus:error:` **will not** be called, but the code does the opposite.

I suspect we *do* want to keep the call to `paymentContext:didFinishWithStatus:error:` at the end, but I'm not sure.